core/src/main/java/org/apache/calcite/adapter/enumerable/EnumUtils.java

```diff
@@ -202,7 +202,23 @@ static Expression fromInternal(Expression e, Class<?> targetType) {
       List<Expression> expressions) {
     final List<Expression> list = new ArrayList<>();
     for (int i = 0; i < expressions.size(); i++) {
-      list.add(fromInternal(expressions.get(i), targetTypes[i]));
+      if (i == targetTypes.length - 1 && targetTypes[i].isArray()) {
+        // only the last type could be varargs. All left expressions
```

Comment:
Should we consider just `Array` type as var-length here ? Should we enumerate all the variable length types ?